File: clinical-appointment-app/src/app/core/models/patient.ts
import { AppointmentListItem } from './appointment';
import { IsoDate, UtcTimestamp } from './primitives';

export interface PatientListItem {
  id: number;
  firstName: string;
  lastName: string;
  fullName: string;
  dateOfBirth: IsoDate;
  phone: string;
  email: string;
  appointmentCount: number;
}

export interface PatientDetail extends PatientListItem {
  createdAt: UtcTimestamp;
  updatedAt: UtcTimestamp;
  appointments: AppointmentListItem[];
}

export interface PatientLookup {
  id: number;
  label: string;
  dateOfBirth: IsoDate;
}

File: backend/src/ClinicalAppointmentSystem.Domain/Entities/Patient.cs
using ClinicalAppointmentSystem.Domain.Common;

namespace ClinicalAppointmentSystem.Domain.Entities;

public class Patient : AuditableEntity
{
    public string FirstName { get; private set; } = string.Empty;

    public string LastName { get; private set; } = string.Empty;

    public DateOnly DateOfBirth { get; private set; }

    public string Phone { get; private set; } = string.Empty;

    public string Email { get; private set; } = string.Empty;

    public ICollection<Appointment> Appointments { get; } = [];

    public string FullName => $"{FirstName} {LastName}";
}
